hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/producers/formats/v6/SidecarWriterV6.java

```diff
@@ -152,6 +159,7 @@ List<SidecarType> types() {
     @Override
     public void close() throws IOException {
         outputStream.close();
+        hashingDelegateStream.close();
```

Comment:
For compressFile=false, outputStream.close() will also close hashingDelegateStream since they're both using fout. This could cause an exception on close due to already closed. Should add a test to verify. Think the solution is not not set hashingDelegateStream when compressFile=false and check if null in close.

Reply:
You would expect this given the normal behavior of `OutputStream` wrapper classes, which propagate a received `close()` to their delegate.

However, `HashingOutputStream` [does not override `close()`](https://github.com/hashgraph/hedera-services/blob/develop/platform-sdk/swirlds-common/src/main/java/com/swirlds/common/crypto/HashingOutputStream.java); and is in fact a no-op.

So what happens with `compressFile=false` is:
 1. `WritableStreamingData` receives `close()`, propagates to `bout`
 2. `BufferedOutputStream` receives `close()`, propagates to `hashingOutputStream`
 3. _<close chaining stops>_
 4. We explicitly call `fout.close()` and now invoke the bottom `FileOutputStream`.

Reply:
Ah, I see. I should've read your code comment.